### DISCUSSION TITLE
Working with CSV files

### DISCUSSION THREAD

**Discussion by profbbrown** [harelba/q#328] [Q&A]
I'm having trouble getting it work with CSV files in which the fields are quoted. Although it separates the fields using the comma delimiter, the values still have the quotes attached to them. Is there a CSV mode for q, that will handle quoted and escaped CSV files correctly?

**Comment by harelba** [comment]
hi, q supports this natively, you can use the `-w` parameter, which allows to choose the input quoting type.

<img width="779" alt="image" src="https://github.com/harelba/q/assets/985765/2e576c4e-f87f-4901-9461-0ebd2171a56c">
